mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1333,11 +1487,35 @@ ParseResult CallOp::parse(OpAsmParser &parser, OperationState &result) {
       return failure();
   }
 
+  auto opBundlesLoc = parser.getCurrentLocation();
+  if (auto result = parseOpBundles(parser, opBundleOperands,
+                                   opBundleOperandTypes, opBundleTags);
+      result.has_value() && failed(*result))
```

Comment:
```suggestion
      result && failed(*result))
```
I think in LLVM/MLIR its more conventional to rely on implicit/contextual bool conversion